src/RawPOSIX/src/safeposix/syscalls/fs_calls.rs

```diff
@@ -1468,8 +1473,22 @@ impl Cage {
                 interface::RustHashEntry::Occupied(mut occupied) => {
                     let segment = occupied.get_mut();
 
-                    segment.unmap_shm(shmaddr, self.cageid);
-
+                    // Use shmdt_handler to unmap and remove from vmmap
```

Comment:
the comments in shmat are more clear than the ones here, if you could add a bit of context that would be good for shmdt